### PR TITLE
eslint: add invenio eslint rules

### PR DESCRIPTION
**Pull request by NRodriguezcuellar**
partly closes https://github.com/inveniosoftware/invenio-app-rdm/issues/1789

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+extends:
+  - '@inveniosoftware/invenio'
+  - '@inveniosoftware/invenio/prettier'
+
+parser: '@babel/eslint-parser'
+
+rules:
+  # allow default prop to be set even if prop required
+  react/default-props-match-prop-types: 0
```
Comment (jrcastro2) [inline .eslintrc.yml:9]: The idea is to not add custom rules to each repo but to use the ones provided by the eslint repo to make sure that all the projects follow the same rules. That said if a rule is meant to be added should be in the [eslint repo](https://github.com/inveniosoftware/eslint-config-invenio). We already defined [here](https://github.com/inveniosoftware/eslint-config-invenio/blob/master/configs/main.yaml#L22) this rule to fail on purpose, why would we want to allow default props for required props?

File: package.json [hunk 2]
```diff
@@ -97,7 +105,7 @@
     "react-invenio-forms": "^0.10.7",
     "react-overridable": "^0.0.3",
     "react-redux": "^7.2.0",
-    "react-scripts": "^3.4.0",
+    "react-scripts": "^5.0.1",
```
Comment (jrcastro2) [inline package.json:108]: Not sure about this changes, we would need to check breaking changes and maybe upgrade it in all other repos like [react-invenio-forms](https://github.com/inveniosoftware/react-invenio-forms/blob/master/package.json#L66). Are you sure you need all this changes in the `package.json`?

Comment (NRodriguezcuellar) [reply package.json:108]: These changes were made because otherwise eslint was failing to lint on my system/editor. Because they are dev dependencies my assumption is that they are only used for development or when building the library, that should make it safe to upgrade (we have some quite old dependencies). What do you think?

File: .eslintrc.yml (hunk 1 above)
Comment (NRodriguezcuellar) [reply .eslintrc.yml:9]: That is a very good point, i did not realize it was set on purpose . I set it here because it seems to be a common pattern for some props and i was not sure about changing the actual code aside from formatting and some minor code refactoring (destructuring prop, adding prop types).

Comment (kpsherva) [reply .eslintrc.yml:9]: on another subject: do we have any specification on the line lenght ? It would be good to align with black(backend formatter)

Comment (jrcastro2) [reply .eslintrc.yml:9]: We use the default value which is 80 if I am not mistaken , but we can update it.

Comment (kpsherva) [reply .eslintrc.yml]: after the discussion, during the stand-up today, I think in the end a prop which has a default is not technically required, since it is not provided when the component is used, so it makes sense to have either default or required, not both

File: package.json (hunk 2 above)
Comment (ntarocco) [reply package.json:108]: This should not be a problem, because they are `devDependencies`: when packaged and released on `npmjs.org`, the modules using this one will not install these dependencies.
What we need always to pay attention is `dependencies` and `peerDeps`.

**Review by ntarocco — COMMENTED** [review]
Should be merged ASAP to avoid infinite amount of conflicts!
Ping @kpsherva @jrcastro2 @zzacharo

File: .eslintrc.yml (hunk 1 above)
Comment (jrcastro2) [reply .eslintrc.yml]: No, not added. We quickly discussed during the stand-up and I personally agree with what @kpsherva explained in the previous comment, the idea is to keep the rule as it is, however this is open to discussion, might be faster IRL.

Comment (NRodriguezcuellar) [reply .eslintrc.yml]: indeed as @jrcastro2 said

File: src/lib/DepositBootstrap.js [hunk 3]
```diff
@@ -158,10 +189,8 @@ const mapDispatchToProps = (dispatch) => ({
   previewAction: (values) => dispatch(preview(values)),
   deleteAction: (values, { isDiscardingVersion }) =>
     dispatch(delete_(values, { isDiscardingVersion })),
-  reservePIDAction: (values, { pidType }) =>
-    dispatch(reservePID(values, { pidType })),
-  discardPIDAction: (values, { pidType }) =>
-    dispatch(discardPID(values, { pidType })),
+  reservePIDAction: (values, { pidType }) => dispatch(reservePID(values, { pidType })),
+  discardPIDAction: (values, { pidType }) => dispatch(discardPID(values, { pidType })),
```
Comment (NRodriguezcuellar) [inline src/lib/DepositBootstrap.js:193]: these should all be required

File: src/lib/components/PublishButton/SubmitReviewModal.js [hunk 4]
```diff
@@ -174,3 +154,7 @@ SubmitReviewModal.propTypes = {
   onSubmit: PropTypes.func.isRequired,
   initialReviewComment: PropTypes.string,
 };
+
+SubmitReviewModal.defaultProps = {
+  initialReviewComment: "",
```
Comment (kpsherva) [inline src/lib/components/PublishButton/SubmitReviewModal.js:159]: will this result in displaying empty comments?

File: src/lib/DepositFormApp.js [hunk 5]
```diff
@@ -112,4 +106,6 @@ DepositFormApp.defaultProps = {
   draftsService: null,
   filesService: null,
   recordSerializer: null,
+  files: undefined,
```
Comment (zzacharo) [inline src/lib/DepositFormApp.js:109]: this can be described more accurately in the Proptypes e.g `Proptypes.shape(...)`

Comment (zzacharo) [reply src/lib/DepositFormApp.js:109]: maybe is better to fallback to `undefined` always instead of null for all the missing props....Do we need to define value for optional properties actually? Can you test if removing them works? I mean the `null`/`undefined`...

File: src/lib/components/Access/EmbargoDateField.js [hunk 6]
```diff
@@ -29,15 +29,17 @@ export function EmbargoDateField({
...
 
 EmbargoDateField.defaultProps = {
-  fieldPath: 'access.embargo.until',
-  label: i18next.t('Embargo until'),
-  labelIcon: 'calendar',
-  placeholder: i18next.t('YYYY-MM-DD'),
+  required: false,
+  labelIcon: "calendar",
+  placeholder: i18next.t("YYYY-MM-DD"),
+  fieldPath: "access.embargo.until",
```
Comment (zzacharo) [inline src/lib/components/Access/EmbargoDateField.js:43]: if we upper pin invenio-app-rdm and we release these changes as `v0.20.x` then we should remove the `defaultProp` `fieldPath`...otherwise, you need here an ignore rule no? I would also maybe add a comment saying that this will be removed in the future or similar...

File: src/lib/components/AccessRightField.js [hunk 7]
```diff
@@ -127,5 +120,8 @@ AccessRightField.propTypes = {
 };
 
 AccessRightField.defaultProps = {
-  fieldPath: 'access',
+  // eslint-disable-next-line react/default-props-match-prop-types
+  fieldPath: "access",
```
Comment (zzacharo) [inline src/lib/components/AccessRightField.js:124]: Check my comment above :)

File: src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js [hunk 8]
```diff
@@ -129,11 +128,3 @@ export class CommunitySelectionSearch extends Component {
     );
   }
 }
-
-CommunitySelectionSearch.propTypes = {
```
Comment (zzacharo) [inline src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js:133]: Why is this removed?

File: src/lib/components/DescriptionsField.js [hunk 9]
```diff
@@ -35,17 +32,19 @@ export class DescriptionsField extends Component {
...
 
 DescriptionsField.defaultProps = {
-  fieldPath: 'metadata.description',
-  label: i18next.t('Description'),
-  labelIcon: 'pencil',
+  // eslint-disable-next-line react/default-props-match-prop-types
+  fieldPath: "metadata.description",
+  label: i18next.t("Description"),
+  labelIcon: "pencil",
   editorConfig: {},
```
Comment (zzacharo) [inline src/lib/components/DescriptionsField.js:48]: maybe use `undefined` here too?

File: src/lib/components/Access/EmbargoDateField.js (hunk 6 above)
Comment (zzacharo) [reply src/lib/components/Access/EmbargoDateField.js:43]: Please take this into account for all the `eslint` ignore rules you added :)

Comment (NRodriguezcuellar) [reply src/lib/components/Access/EmbargoDateField.js:44]: Yes, i looked over this one, i just made sure that every `fieldPath` prop is set to `isRequired`!

Comment (NRodriguezcuellar) [reply src/lib/components/Access/EmbargoDateField.js:44]: maybe i should post a comment in this PR documenting why this decision (ignoring the fieldPaths defaultProps) was made and link it with every ignore?

File: src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js (hunk 8 above)
Comment (NRodriguezcuellar) [reply src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js:133]: because the components uses no props 😄

File: src/lib/DepositFormApp.js (hunk 5 above)
Comment (NRodriguezcuellar) [reply src/lib/DepositFormApp.js:109]: Yes it is a rule that every propType that is not `required` should have a defaultProp, as you see i usually default to undefined so that the behaviour does not change unexpectedly.

as for setting the shape i am not sure what that could be in this case 😅

File: src/lib/components/Access/EmbargoDateField.js (hunk 6 above)
Comment (kpsherva) [reply src/lib/components/Access/EmbargoDateField.js:44]: I would start an issue to clean it out in the 0.20 maybe

Comment (NRodriguezcuellar) [reply src/lib/components/Access/EmbargoDateField.js:44]: issue made here https://github.com/inveniosoftware/react-invenio-deposit/issues/563

**Review by kpsherva — APPROVED** [review]
